### PR TITLE
Added internal-grpc-port to standalone run options

### PR DESCRIPTION
**Pull request by jigargandhi**
Signed-off-by: Jigar R Gandhi <jigarr.gandhi@gmail.com>

# Description

Added dapr-internal-grpc-port argument to daprd process command line for standalone.
This value will be used when parsing the value from a local DNS component which is implemented as per this
https://github.com/dapr/components-contrib/pull/1340.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR related to: dapr/dapr#3256

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by mukundansundar** [comment]
@yaron2 Can you review this PR?

**Comment by jigargandhi** [comment]
@yaron2  please review this

**Comment by mukundansundar** [comment]
@jigargandhi Please resolve the conflicts in the PR ...

cc @yaron2 once conflicts are resolved, please review the PR

**Comment by jigargandhi** [comment]
@yaron2 please review the PR, @mukundansundar I have resolved the merge conflicts

**Comment by mukundansundar** [comment]
@jigargandhi Can you resolve the conflicts for this?

**Comment by mukundansundar** [comment]
@jigargandhi Can you resolve the conflicts for this PR?

**Comment by jigargandhi** [comment]
@mukundansundar resolved the conflicts

**Comment by mukundansundar** [comment]
@pravinpushkar Please review this PR ..

**Comment by pravinpushkar** [comment]
@jigargandhi FYI - The related PRs in the description are not merged and probably got closed due to no activity.

**Comment by pravinpushkar** [comment]
@jigargandhi Does it make sense if we expose this as configurable option (get free port only when not provided by cmd line)? Another clarification, if this  is supposed to be only internal then should it be part of cli ?

File: cmd/run.go [hunk 1]
```diff
@@ -372,3 +379,18 @@ func init() {
 
 	RootCmd.AddCommand(RunCmd)
 }
+
+// GetFreePort asks the kernel for a free open port that is ready to use.
+func getFreePort() (int, error) {
```
Comment (mukundansundar) [inline cmd/run.go:384]: @jigargandhi we are already importing `"github.com/phayes/freeport"` in the `pkg/standalone/run.go` file. Why can't we use that here?

File: cmd/run.go [hunk 2]
```diff
@@ -99,6 +100,11 @@ dapr run --app-id myapp --app-port 3000 --app-protocol grpc -- go run main.go
 			}
 		}
 
+		internalGrpcPort, err := getFreePort()
```
Comment (mukundansundar) [inline cmd/run.go:103]: Can you also change the implementation to allow passing in internal grpc port from cmd flag? Wrt issue #1026

**Review by mukundansundar — COMMENTED** [review]
@jigargandhi Thanks for the contribution!
Also please do add E2E for this.

**Comment by jigargandhi** [comment]
I am not planning to work on this at the moment, please feel free to take it up.

**Comment by shubham1172** [comment]
I can take this PR to completion, will cherry pick commits and create a new PR.

**Comment by mukundansundar** [comment]
closed by #1040

**Pull request closed without merging**